WIP: Fix truncated text on pages

kr8n3r opened the pull request:
## What 
Remove truncated text from our pages by removing CSS styles that prevent the text to wrap and associated HTML classes. Text in such scenarios now flows and wraps. 

## Why
Content must be presented without loss of information or functionality.
This fails [the reflow WCAG success criterion](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html).

## How to review
- see individual commits

These changes affect many pages that include affected components. 
- `file-list` component
   - services/serviceID>/inbox
   - services/serviceID>/notifications
   - services/serviceID>/uploads
   - services/serviceID>unsubscribe-requests
   - services/serviceID>/contact-list/<id>
   - services/serviceID>/api/keys
   - services/serviceID>/jobs/<id>
   - send-contact-list, returned-letters
- `dashboard-table`
   - services/serviceID>api/callback
   - services/serviceID>template-usage
   - /features/performance
   - notifications table
- settings summary list
   - service settings
   - organisation settings
- template folder path heading
  - a template in nested folder: services/serviceID>templates/id
  
  ## Visual changes
  
  There are changes in listed page that, instead of a single line of truncated text, now show all the text that wraps into multiple line as required. 
  
  ### Example: Notifications table
  

https://github.com/user-attachments/assets/dd6dafe0-c673-4dd7-9185-5f61a0f8cb72


https://github.com/user-attachments/assets/6c5ebb4a-1c5e-457e-991f-7f53d4d50ba1

kr8n3r committed:
Remove text truncation from the table macro

We should no longer have any trucated content in
tables, so best to remove that option from the
macro.

Additionally, remove associated CSS styles.

kr8n3r committed:
Remove truncated text from settings pages

Remove CSS styles and classes that were added
to the govukSummaray component on service settings
and organisation settings pages.

kr8n3r committed:
Remove text truncation from template folder-path heading

When viewing a template we now show all the text
for it's path.

before:
T... > [icon] > [icon] lv2 > templateName

after:
Templates > [icon] lv1  > [icon] lv2  > templateName

tombye reviewed (changes requested):
Good, clear, work that touches a lot of our codebase to nice one for finding all the bits of each type of truncation.

I've left a few comments but, overall, I'm interested to know if we should just be swapping out our current clipping of text with an ellipsis for the gradual breaking behaviour you get from `word-break: break-word` (see `overflow-wrap: break-word` section on https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap for that, which is the same).

kr8n3r committed:
Remove text truncation from filelist component

Remove overflow hidden, text-overflow and
space no-wrap.
We want all the content to be displayed and
reflow naturally inside the containers.

Remove block link focus styles
As text can wrap into multiple lines we can no
longer predictably set styles that would
conveniently cover the entire container height.
Now only the link retains the focus style, as
per Design System.

Break word on long emails

kr8n3r committed:
Remove text truncation from table component

Renove forced truncation and custom focus styles
as block focus can no longer be used due to text
wrapping into multiple lines. Now, there is no need
to adjust the hint text colour.

From affects the table on
- unsubscribe requests, template statistics. jobs,
performance, callbacks and notifications pages

kr8n3r commented on app/assets/stylesheets/views/dashboard.scss:
```diff
@@ -154,22 +136,12 @@
-.govuk-link:focus {
+// Note: this selector should use the :has() pseudo-class in the future when it is supported
+//       which will remove the need for JS
+.file-list .js-child-has-focus {
```
👍 done

kr8n3r commented on app/assets/stylesheets/components/table.scss:
```diff
@@ -39,13 +39,6 @@
       width: 52.5%;
       font-weight: normal;
-
-      .hint,
```
added

kr8n3r commented on app/assets/stylesheets/views/dashboard.scss:
```diff
@@ -95,25 +95,12 @@
   &-filename-large {
     @include govuk-font(24, $weight: bold);
     display: block;
-    white-space: nowrap;
```
👍 done

kr8n3r committed:
Introduce block focus

tombye commented on app/assets/stylesheets/views/dashboard.scss:
```diff
@@ -154,22 +136,12 @@
+//       which will remove the need for JS
+.file-list .js-child-has-focus {
 
-  &.file-list-filename,
```
We've been discussing this outside of this pull request and Jani's since been trialing this approach so marking this as resolved.

kr8n3r commented:
closing for now in favour of https://github.com/alphagov/notifications-admin/pull/5288 which has a smaller scope.
will keep the branch for as while
